cdap-watchdog/src/main/java/co/cask/cdap/logging/write/FileMetaDataManager.java

```diff
@@ -264,6 +271,13 @@ public String call() throws Exception {
                   LOG.debug("Got file {} with start time {}", file, Bytes.toLong(colName));
                 }
 
+                if (Strings.isNullOrEmpty(namespacedLogDir)) {
+                  LOG.warn("Delete metadata for {} namespace", namespaceId.getEntityName());
+                  table.delete(rowKey, colName);
+                  deletedColumns++;
+                  continue;
+                }
+
                 Location fileLocation = impersonator.doAs(namespaceId, new Callable<Location>() {
```

Comment:
The `for` loop has try-catch block. So any exceptions in that block should not terminate the transaction. However, it will keep the metadata entry as is. Other way to handle this would be to remove meta entry when we see `NamespaceNotFoundException`.